cfme/common/__init__.py

```diff
@@ -258,6 +265,26 @@ def step(self):
         self.prerequisite_view.toolbar.policy.item_select('Edit Tags')
 
 
+@navigator.register(WidgetasticTaggable, 'EditTags')
+class EditTagsFromList(CFMENavigateStep):
+    VIEW = TagPageView
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        self.prerequisite_view.entities.get_entity(surf_pages=True, name=self.obj.name).check()
+        self.prerequisite_view.toolbar.policy.item_select('Edit Tags')
+
+
+@navigator.register(WidgetasticTaggable, 'EditTagsCollection')
+class EditTagsFromListCollection(CFMENavigateStep):
+    VIEW = TagPageView
+    prerequisite = NavigateToAttribute('parent', 'All')
+
+    def step(self):
```

Comment:
**Question**

Thought about this design for a while. I do kind of like registering the nav destinations against the inheritable class.
I think that we need to have a kwarg passed to this destination's step though, a list of entities in the collection.

I'm considering a single navigation class, with a prerequisite that uses `isinstance(self.obj, BaseCollection)` to select the prerequisite call.  `step()` taking `**kwargs` would allow for optionally passing a list of entities to check.